.github: Migrate ecr_gc to github actions

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #62286
* __->__ #62284

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29943356](https://our.internmc.facebook.com/intern/diff/D29943356)

seemethere committed:
.github: Migrate ecr_gc to github actions

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29943356).

seemethere commented on .github/workflows/ecr_gc_docker.yml:
```diff
@@ -0,0 +1,43 @@
+  workflow_dispatch:
+
+env:
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ECR_GC_ACCESS_KEY_ID }}
```
I've already went through and added these secrets here

seemethere commented on .circleci/ecr_gc_docker/gc.py:
```diff
@@ -7,6 +7,69 @@
 
+# TODO: See if this is actually still needed
+# Default list of legacy tags to keep
+IGNORE_TAGS = {
```
I'm not entirely sure if we actually need these anymore

zhouzhuojie commented on .github/workflows/ecr_gc_docker.yml:
```diff
@@ -0,0 +1,43 @@
+          echo "export GENERATED_IMAGE_TAG='${GENERATED_IMAGE_TAG}'" >> "${BASH_ENV}"
+      - name: Run garbage collection
+        run: |
+          .circleci/ecr_gc_docker/gc.py --filter-prefix "${FILTER_PREFIX}" --ignore-tags "${GENERATED_IMAGE_TAG}"
```
oh nice, get rid of the gc image all together! 🎉

malfet reviewed (approved):
LGTM, though I wonder if we shoudl also delete `.circleci/verbatim-sources/workflows/workflows-ecr-gc.yml` as well part of `.circleci/verbatim-sources/job-specs/docker_jobs.yml`

seemethere commented:
> LGTM, though I wonder if we shoudl also delete `.circleci/verbatim-sources/workflows/workflows-ecr-gc.yml` as well part of `.circleci/verbatim-sources/job-specs/docker_jobs.yml`

We can delete them in a follow up after we verify the workflow actually works

zhouzhuojie commented on .github/workflows/ecr_gc_docker.yml:
```diff
@@ -0,0 +1,43 @@
+    # Runs every 30 minutes
+    - cron: "*/30 * * * *"
+  # Have the ability to trigger this job manually using the API as well
+  workflow_dispatch:
```
is it green when you manually triggering it?

seemethere replied:
Can't actually manually trigger it until we merge first unfortunately.

Would be a good use for a canary repository though

zhouzhuojie replied:
I think we can add on pull_request temporally to test the script, and then before the merge remove the on pull_request trigger

seemethere committed:
Update on ".github: Migrate ecr_gc to github actions"


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29943356](https://our.internmc.facebook.com/intern/diff/D29943356)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29943356).
